src/libraries/System.Text.Json/tests/Common/JsonSchemaExporterTests.TestTypes.cs

```diff
@@ -0,0 +1,1320 @@
+            yield return new TestData<byte[]>(
+                Value: [1, 2, 3],
+                AdditionalValues: [[]],
+                ExpectedJsonSchema: """{"type":["string","null"]}""");
+
+            yield return new TestData<Memory<byte>>(new byte[] { 1, 2, 3 }, ExpectedJsonSchema: """{"type":"string"}""");
+            yield return new TestData<ReadOnlyMemory<byte>>(new byte[] { 1, 2, 3 }, ExpectedJsonSchema: """{"type":"string"}""");
+            yield return new TestData<DateTime>(
+                Value: new(2024, 06, 06, 21, 39, 42, DateTimeKind.Utc),
+                ExpectedJsonSchema: """{"type":"string","format":"date-time"}""");
+
+            yield return new TestData<DateTimeOffset>(
+                Value: new(new DateTime(2021, 1, 1), TimeSpan.Zero),
+                AdditionalValues: [DateTimeOffset.MinValue, DateTimeOffset.MaxValue],
+                ExpectedJsonSchema: """{"type":"string","format": "date-time"}""");
+
+            yield return new TestData<TimeSpan>(
+                Value: new(hours: 5, minutes: 13, seconds: 3),
+                AdditionalValues: [TimeSpan.MinValue, TimeSpan.MaxValue],
+                ExpectedJsonSchema: """{"type":"string", "pattern": "^-?(\\d+\\.)?\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,7})?$"}""");
+
+#if NET
+            yield return new TestData<DateOnly>(new(2021, 1, 1), ExpectedJsonSchema: """{"type":"string","format": "date"}""");
+            yield return new TestData<TimeOnly>(new(hour: 22, minute: 30, second: 33, millisecond: 100), ExpectedJsonSchema: """{"type":"string","format": "time"}""");
+#endif
+            yield return new TestData<Guid>(Guid.Empty, ExpectedJsonSchema: """{"type":"string","format":"uuid"}""");
+            yield return new TestData<Uri>(new("http://example.com"), """{"type":["string","null"],"format":"uri"}""");
+            yield return new TestData<Version>(new(1, 2, 3, 4), ExpectedJsonSchema: """{"type":["string","null"],"pattern":"^\\d+(\\.\\d+){1,3}$"}""");
+            yield return new TestData<JsonDocument>(JsonDocument.Parse("""[{ "x" : 42 }]"""), ExpectedJsonSchema: "true");
+            yield return new TestData<JsonElement>(JsonDocument.Parse("""[{ "x" : 42 }]""").RootElement, ExpectedJsonSchema: "true");
+            yield return new TestData<JsonNode>(JsonNode.Parse("""[{ "x" : 42 }]"""), ExpectedJsonSchema: "true");
+            yield return new TestData<JsonValue>((JsonValue)42, ExpectedJsonSchema: "true");
+            yield return new TestData<JsonObject>(new() { ["x"] = 42 }, ExpectedJsonSchema: """{"type":["object","null"]}""");
+            yield return new TestData<JsonArray>([(JsonNode)1, (JsonNode)2, (JsonNode)3], ExpectedJsonSchema: """{"type":["array","null"]}""");
+
+            // Enum types
+            yield return new TestData<IntEnum>(IntEnum.A, ExpectedJsonSchema: """{"type":"integer"}""");
+            yield return new TestData<StringEnum>(StringEnum.A, ExpectedJsonSchema: """{"type":"string","enum":["A","B","C"]}""");
```

Comment:
`type: string` here is redundant.  `enum` specifies the exact values that are accepted.

Reply:
For context, this was added here https://github.com/eiriktsarpalis/stj-schema-mapper/pull/1 in reaction to the Gemini API requiring a `type` keyword for enums. In the interest of correctness it might be possible to remove the keyword here and then work with the Semantic Kernel team so that this is added back via a custom callback.

cc @stephentoub @markwallace-microsoft

Reply:
I've decided to remove the type keyword from enum types, but added a test demonstrating how this can be added back using a custom callback.